extensions/tools/scripts/creeps.js

```diff
@@ -10,6 +10,7 @@ module.exports = function() {
 
         if (typeof creep.memory.role !== "string") {
             console.log('Warning: Creep without role at ' + creep.pos.x + "," + creep.pos.y);
+            creep.memory.role = creep.name.split(' ')[0];
```

Comment:
Maybe I'll let a hook handle this, so the behavior can be modified or overwritten.